### PR TITLE
docs: add Web Storage server-side API documentation

### LINKED ISSUE #2518: Add server-side API for Web Storage
Add a documentation for server-side API for Web Storage.
See this pull request https://github.com/vaadin/flow/pull/16978 .

If you think this issue is important, add a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to help the community and maintainers prioritize this issue.

### PR DESCRIPTION
**Pull request by tltv**
Closes #2518

### CODE REVIEW

**Review by Mikaelsu — APPROVED** [review]

File: frontend/themes/docs/theme-editor.css
Comment (jouni) [inline frontend/themes/docs/theme-editor.css:1]: Accidentally added, please remove.

File: package-lock.json
Comment (jouni) [inline package-lock.json:1]: This is likely also accidentally updated, please revert.

File: articles/advanced/web-storage-api.asciidoc [hunk 1]
```diff
@@ -79,6 +81,6 @@ WebStorage.getItem(Storage.SESSION_STORAGE, "key", value -> {
 });
 ----
 
-[NOTE]
 Callback gives a null value in case of an unexpected error.
 
+[discussion-id]`08b13e40-3845-11ee-be56-0242ac120002`
```
Comment (jouni) [inline articles/advanced/web-storage-api.asciidoc:86]: Nicely remembered 👍

File: frontend/themes/docs/theme-editor.css
Comment (jouni) [reply frontend/themes/docs/theme-editor.css:1]: Done in https://github.com/vaadin/docs/commit/73a7eb04ed13245058cf62b2d8d3be223f38d0d8

File: package-lock.json
Comment (jouni) [reply package-lock.json:1]: Done in https://github.com/vaadin/docs/commit/73a7eb04ed13245058cf62b2d8d3be223f38d0d8

**Pull request merged**